fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapper.kt

```diff
@@ -0,0 +1,67 @@
+package org.mozilla.fenix.shopping.middleware
+
+import mozilla.components.browser.engine.gecko.shopping.GeckoProductAnalysis
+import mozilla.components.browser.engine.gecko.shopping.Highlight
+import mozilla.components.concept.engine.shopping.ProductAnalysis
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState.OptedIn.ProductReviewState
+
+/**
+ * Maps [ProductAnalysis] to [ProductReviewState].
+ */
+fun ProductAnalysis?.toProductReviewState(): ProductReviewState =
+    if (this == null) {
+        ProductReviewState.Error
+    } else {
+        when (this) {
+            is GeckoProductAnalysis -> toProductReview()
+            else -> ProductReviewState.Error
+        }
+    }
+
+private fun GeckoProductAnalysis.toProductReview(): ProductReviewState =
+    if (productId == null) {
+        ProductReviewState.Error
+    } else {
+        val mappedRating = adjustedRating.toFloatOrNull()
+        val mappedGrade = grade?.toGrade()
+        val mappedHighlights = highlights?.toHighlights()?.toSortedMap()
+
+        if (mappedGrade == null && mappedRating == null && mappedHighlights == null) {
+            ProductReviewState.NoAnalysisPresent
+        } else {
+            ProductReviewState.AnalysisPresent(
+                productId = productId!!,
+                reviewGrade = mappedGrade,
+                needsAnalysis = needsAnalysis,
+                adjustedRating = mappedRating,
+                productUrl = analysisURL!!,
+                highlights = mappedHighlights,
+            )
+        }
+    }
+
+private fun String.toGrade(): ReviewQualityCheckState.Grade? =
+    try {
+        ReviewQualityCheckState.Grade.valueOf(this)
+    } catch (e: IllegalArgumentException) {
+        null
+    }
+
+private fun Highlight.toHighlights(): Map<ReviewQualityCheckState.HighlightType, List<String>>? {
+    val highlights: Map<ReviewQualityCheckState.HighlightType, List<String>?> = mapOf(
+        ReviewQualityCheckState.HighlightType.QUALITY to quality,
+        ReviewQualityCheckState.HighlightType.PRICE to price,
+        ReviewQualityCheckState.HighlightType.SHIPPING to shipping,
+        ReviewQualityCheckState.HighlightType.COMPETITIVENESS to competitiveness,
```

Comment:
is the omission of `PACKAGING_AND_APPEARANCE` intentional?

Reply:
Nope, great catch! Fixed! Incidentally the test was also setting it to null to test the case for missing highlights, added another test with all highlights to make sure this is fixed.

Reply:
Sorry a little late on this. Wondering if it would be useful here to use the `HightlightType` enum to enforce all values are used e.g.

``` 
val highlights: Map<ReviewQualityCheckState.HighlightType, List<String>?> =
        ReviewQualityCheckState.HighlightType.values().associate { type ->
            when (type) {
                ReviewQualityCheckState.HighlightType.QUALITY -> type to quality
                ReviewQualityCheckState.HighlightType.PRICE -> type to price
                ReviewQualityCheckState.HighlightType.SHIPPING -> type to shipping
                ReviewQualityCheckState.HighlightType.PACKAGING_AND_APPEARANCE -> type to appearance
                ReviewQualityCheckState.HighlightType.COMPETITIVENESS -> type to competitiveness
            }
        }
```
then the IDE would show any missing types

Reply:
No worries, thanks! This is a good idea to make sure the IDE/compiler can assist us. I've updated this in #3212